packages/algoliasearch-helper/src/utils/sortAndMergeRecommendations.js

```diff
@@ -21,20 +21,24 @@ function getAverageIndices(indexTracker, nrOfObjs) {
   });
 }
 
-function sortAndMergeRecommendations(results) {
+function sortAndMergeRecommendations(objectIDs, results) {
   var indexTracker = {};
 
   results.forEach(function (hits) {
-    hits.forEach(function (hit, index) {
-      if (!indexTracker[hit.objectID]) {
-        indexTracker[hit.objectID] = { indexSum: index, count: 1 };
-      } else {
-        indexTracker[hit.objectID] = {
-          indexSum: indexTracker[hit.objectID].indexSum + index,
-          count: indexTracker[hit.objectID].count + 1,
-        };
-      }
-    });
+    hits
+      .filter(function (hit) {
+        return !objectIDs.includes(hit.objectID);
+      })
```

Comment:
you can put this inside the forEach and early return, that's at least one iteration less